build_tools/spec/custom_service_spec.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'spec_helper'
+
+module BuildTools
+
+  describe 'CustomService' do
+
+    it 'keeps svc_name camel case' do
+      cs = CustomService.new(
+        service_name: "SDkTestApi",
+        default_endpoint: "https://foo.com",
+        model_path: "foo/path"
+      )
+      expect(cs.svc_name).to eq("SDkTestApi")
+    end
+
+    it 'decode svc_name properly' do
+      cs = CustomService.new(
+        service_name: "Sdk%20%20test%20Api", 
+        default_endpoint: "https://foo.com",
+        model_path: "foo/path"
+      )
+      expect(cs.svc_name).to eq("SdkTestApi")
+    end
+
```

Comment:
I'd actually like us to add more tests. For example, service names with spaces, and possibly some service names which should fail by intention. This covers the basics, I'd like to get a bit creative on cases.

Reply:
Added more tests in the latest commit, unfortunately, it would be tricker for failed scenarios since  that will happened at Kernel require time later at build times